include/pybind11/pybind11.h

```diff
@@ -1164,7 +1168,7 @@ iterator make_key_iterator(Iterator first, Iterator last, Extra &&... extra) {
                     s.first = false;
                 if (s.it == s.end)
                     throw stop_iteration();
-                return s.it->first;
```

Comment:
Just one less requirement for the iterators, in general they're not required to implement `->`.